libsdl-org/SDL_image
# cmake: fix test installation + fix(IMG_INIT_WEBP) + add -Wall

- tests were not installed
- fix test environment
- Fixes #324, should be backported to SDL2
- add `-Wall -Wextra` compile options
- add `-DSDL3IMAGE_WERROR` cmake option to optionally add `-Werror`/`-WX`
- don't do `find_package(SDL3)` when a SDL3 target is available

After this pr:
- `testimage` fails when loading `sample.xcf`.
- the addition of `-Wall` `-Wextra` causes extra warnings (see below) =>fixed in 9ccae48 (please review)


```
/home/maarten/programming/SDL_image/IMG_xcf.c: In function ‘load_xcf_tile_none’:
/home/maarten/programming/SDL_image/IMG_xcf.c:539:36: warning: comparison of integer expressions of different signedness: ‘Sint64’ {aka ‘long int’} and ‘Uint64’ {aka ‘long unsigned int’} [-Wsign-compare]
  539 |     if (SDL_RWread(src, load, len) != len) {
      |                                    ^~
/home/maarten/programming/SDL_image/IMG_xcf.c: In function ‘load_xcf_tile_rle’:
/home/maarten/programming/SDL_image/IMG_xcf.c:560:32: warning: comparison of integer expressions of different signedness: ‘Sint64’ {aka ‘long int’} and ‘Uint64’ {aka ‘long unsigned int’} [-Wsign-compare]
  560 |   if ( SDL_RWread(src, t, len) != len ) {
      |                                ^~
[20/29] Building C object CMakeFiles/SDL3_image.dir/IMG_xpm.c.o
/home/maarten/programming/SDL_image/IMG_xpm.c: In function ‘create_colorhash’:
/home/maarten/programming/SDL_image/IMG_xpm.c:119:48: warning: comparison of integer expressions of different signedness: ‘long unsigned int’ and ‘int’ [-Wsign-compare]
  119 |     if ((bytes / sizeof(struct hash_entry **)) != hash->size) {
      |                                                ^~
/home/maarten/programming/SDL_image/IMG_xpm.c:132:45: warning: comparison of integer expressions of different signedness: ‘long unsigned int’ and ‘int’ [-Wsign-compare]
  132 |     if ((bytes / sizeof(struct hash_entry)) != maxnum) {
      |                                             ^~
/home/maarten/programming/SDL_image/IMG_xpm.c: In function ‘color_to_argb’:
/home/maarten/programming/SDL_image/IMG_xpm.c:906:23: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
  906 |         for (i = 0; i < SDL_arraysize(known); i++) {
      |                       ^
/home/maarten/programming/SDL_image/IMG_xpm.c: In function ‘load_xpm’:
/home/maarten/programming/SDL_image/IMG_xpm.c:1042:37: warning: comparison of integer expressions of different signedness: ‘long unsigned int’ and ‘int’ [-Wsign-compare]
 1042 |     if ((size_t)(ncolors * cpp)/cpp != ncolors) {
      |                                     ^~
```

Fixes #326
